### PR TITLE
Checkfile for IntegralSolverPySCF

### PR DESCRIPTION
**Pull request by AlexandreF-1qbit**
Support for defining a chkfile during the mean-field calculation. When providing a chkfile name to the `IntegralSolverPySCF` class (`IntegralSolverPySCF("your_chkfile_name.chk")`), it will consider using a file on disk to store mo coefficients / density matrix and other import quantities.

### CODE REVIEW

File: tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py [hunk 1]
```diff
@@ -108,14 +111,21 @@ def compute_mean_field(self, sqmol):
         sqmol.mean_field = self.scf.RHF(molecule) if not sqmol.uhf else self.scf.UHF(molecule)
         sqmol.mean_field.verbose = 0
 
+        if self.chkfile is not None:
```
Comment (ValentinS4t1qbit) [inline tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py:114]: `is_chkfile_exist` -> `chkfile_found`

Suggestion: define variable with default value out of branch. Update if necessary.

```
chkfile_found = False
if self.chkfile:
   chkfile_found = os.path.exists(self.chkfile)
   sqmol.mean_field.chkfile = self.chkfile
```

File: tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py [hunk 2]
```diff
@@ -108,14 +111,21 @@ def compute_mean_field(self, sqmol):
...
         # Force broken symmetry for uhf calculation when spin is 0 as shown in
         # https://github.com/sunqm/pyscf/blob/master/examples/scf/32-break_spin_symm.py
-        if sqmol.uhf and sqmol.spin == 0:
+        if sqmol.uhf and sqmol.spin == 0 and not is_chkfile_exist:
             dm_alpha, dm_beta = sqmol.mean_field.get_init_guess()
             dm_beta[:1, :] = 0
             dm = (dm_alpha, dm_beta)
             sqmol.mean_field.kernel(dm)
         else:
+            sqmol.mean_field.init_guess = "chkfile" if is_chkfile_exist else "minao"
```
Comment (ValentinS4t1qbit) [inline tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py:128]: So, if a user provides a path to a check file that is incorrect or does not exist, you're not going to tell them or raise an error, instead you're gonna do computations and make them believe everything went according to their plan ?

I'd be tempted to raise a warning or an error as soon as I realize the check file cannot be found. Do you think computing from scratch if it cannot be found is desirable behaviour for users who explicitly tried to provide a chkfile ? In what cases ?

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
I'm not sure about how we should handle this if the chkfile is not found.

File: tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py (hunk 2 above)
Comment (AlexandreF-1qbit) [reply tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py:128]: If there is no chkfile already existing, the code will create one (and the initial guess stays the default one). Without that, the users wouldn't be able to create a chkfile, preventing them to use the feature of getting initial guesses from this file for future calculations. It is actually the expected behaviour.

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
We need a bit of docs with your explanations from the PR conversation tab (which you can resolve afterwards). Then I believe we are good to merge !

File: tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py [hunk 3]
```diff
@@ -44,13 +46,14 @@ def mol_to_pyscf(mol, basis="CRENBL", symmetry=False, ecp=None):
 
 class IntegralSolverPySCF(IntegralSolver):
     """Electronic Structure integration for pyscf"""
-    def __init__(self):
+    def __init__(self, chkfile=None):
```
[the remark this answers is not in the document]
Comment (AlexandreF-1qbit) [reply tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py:50]: A docstring should be there now.

**Review by ValentinS4t1qbit — APPROVED** [review]
Thank you ! Please think about whether this feature should be advertised or showcased in a way or another. If not particularly, that's fine

**Pull request merged**